### ISSUE TITLE
Missing Consent for MicrosoftDefenderAdvancedThreatProtection Data Connector Enablement.

### ISSUE THREAD

**Issue by alliallfrey** [Azure/Azure-Sentinel#5007]
The data connector 'Microsoft Defender for Endpoint`, labelled as 'MicrosoftDefenderAdvancedThreatProtection' within https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Sentinel-All-In-One/Powershell/SentinelallInOne.ps1, does not run successfully. The error is: {
  "error": {
    "code": "InvalidLicense",
    "message": "Missing consent"
  }
}

This issue occurs with the New-AzSentinelDataConnector PowerShell command, the ARM template for data connectors and the API directly.

Using the same account (and same permissions) to enable this data connector via the portal works fine. 

<img width="196" alt="image" src="https://user-images.githubusercontent.com/11616100/169276796-5eb4856b-c2da-49e2-a047-77105a7dfb2e.png">

The same issue is also posted here, https://github.com/Azure/SimuLand/issues/23, with no resolution.

Any help greatly appreciated, thanks.

**Comment by javiersoriano** [comment]
@alliallfrey I checked the ARM template version of Sentinel All-In-One and also this template here: https://github.com/Azure/Azure-Sentinel/tree/master/Tools/ARM-Templates/DataConnectors/MDATP and both work fine to enable the connector. Could you double check?

I did not try the PS method though

**Comment by alliallfrey** [comment]
@javiersoriano Thank you for looking into this. Unfortunately I get the same issue when I use https://github.com/Azure/Azure-Sentinel/tree/master/Tools/ARM-Templates/DataConnectors/MDATP

My bicep template deploys in the same way as MDATP.json, and have tried all API versions available for https://docs.microsoft.com/en-us/azure/templates/microsoft.securityinsights/dataconnectors?tabs=bicep

**Comment by javiersoriano** [comment]
did you try in different environments and different permissions? As I said above, I can't reproduce your issue with a user with the appropriate permissions.

**Comment by alliallfrey** [comment]
I have tried it with different accounts and different permissions: 
  Defender for Endpoint: 
    - Global Administrator
    - Security Administrator
  Sentinel Workspace:
    - Microsoft Sentinel Contributor
    - Owner
I have also tried different environments, though these have different licenses available:
  Environment 1 using Microsoft 365 E5
  Environment 2 using Microsoft Defender for Endpoint P2 (Trial)

Within environment 1 I can enable the connector via the portal, so I don't think the user or permissions are the issue.
Within environment 2 I can't enable the connector via the portal because the license prerequisite fails. Hopefully I haven't missed something obvious with this.

**Comment by SewageWorker** [comment]
I am also experiencing the same issue. Have tried different API's etc. I have been trying to deploy via Terraform, then using ARM templates...

**Comment by javiersoriano** [comment]
@alliallfrey from your last message, it seems like the issue is license and not permissions. Your license should be in this list: https://docs.microsoft.com/en-us/microsoft-365/security/defender/prerequisites?view=o365-worldwide#licensing-requirements

Seems like Trial offers are not enough. 

@SewageWorker what 365 Defender license do you have? (see above for supported ones)

**Comment by alliallfrey** [comment]
> @alliallfrey from your last message, it seems like the issue is license and not permissions. Your license should be in this list: https://docs.microsoft.com/en-us/microsoft-365/security/defender/prerequisites?view=o365-worldwide#licensing-requirements
> 
> Seems like Trial offers are not enough.
> 
> @SewageWorker what 365 Defender license do you have? (see above for supported ones)

**Comment by alliallfrey** [comment]
Thank you for following up @javiersoriano, the main tenant I am using has Microsoft 365 E5, not trial.

Also apologies for closing this, accidently clicked the wrong button!

**Comment by javiersoriano** [comment]
Right, @alliallfrey , but you mentioned above that for the environment with E5, the connector onboarding was working properly. It was not working for the other environment where you had trial, correct?

**Comment by SewageWorker** [comment]
@javiersoriano At the moment we are running E5 Security Trail on top of a running E3 subscription. Just purchasing a full E5 Security license then will try it again and report back on if that works

**Comment by alliallfrey** [comment]
@javiersoriano Partially correct... The connector onboarding works via the portal, but does not via an API call. This behaviour is with the same user/permissions.

**Comment by SewageWorker** [comment]
@javiersoriano Still getting the same error message even with a paid license and not a trial one

**Comment by javiersoriano** [comment]
@SewageWorker did you try with PowerShell or [ARM template](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/ARM-Templates/DataConnectors/MDATP)?

**Comment by SewageWorker** [comment]
@javiersoriano Yes i did and unfortunately still the same

**Comment by javiersoriano** [comment]
can't reproduce

**Comment by noapocalypse** [comment]
@alliallfrey Hey, did you ever work out the issue with this? We are getting the same error when deploying from ARM
